src/App.jsx

```diff
@@ -14,33 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_NAME = 'goodName';
+const SORT_LENGTH = 'goodLength';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getVisibleGoods(goods, { sortField, reversed }) {
+  let res = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    switch (sortField) {
+      case SORT_NAME:
+        res = res.sort();
+        break;
+      case SORT_LENGTH:
+        res = res.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        break;
+    }
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
+    res.reverse();
+  }
+
+  return res;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getVisibleGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
+  const reset = () => {
+    setSortField('');
+    setReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_NAME,
+          })}
+          onClick={() => setSortField(SORT_NAME)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SORT_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', { 'is-light': !reversed })}
+          onClick={() => {
+            setReversed(!reversed);
+          }}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reversed) && (
```

Comment:
move on to var `const isResetBtnVisible = sortField || reversed;` and just  `isResetBtnVisible && (`